File: Algo/Leetcode/122BestTimeToBuyAndSellStockII.cpp
// Calculate profit between every two days
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_v = 0;
        for(int i=1; i<prices.size();i++){
            max_v += max(0, prices[i] - prices[i-1]);
        }
        return max_v;
    }
};

File: Algo/Leetcode/122BestTimeToBuyAndSellStockII.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        n = len(prices)
        res = 0
        for i in range(n - 1):
            res += max(0, prices[i + 1] - prices[i])

        return res